### PR TITLE
diamond-improvements

### PR DESCRIPTION
**Pull request by idob-cloudify**
three improvements:
1) reduce log rotation from 7 days to 2 days
2) reduce log spamming when there is long lasting connection error
3) init.d start command to run as user and not root

### CODE REVIEW

File: diamond_agent/resources/diamond [hunk 1]
```diff
@@ -26,7 +26,7 @@ case "$1" in
     start)
     export CELERY_WORK_DIR="{{ WORK_DIR }}"
     export CLOUDIFY_CLUSTER_SETTINGS_PATH="{{ CLUSTER_SETTINGS_PATH }}"
-    {{ CMD }}
+    sudo su {{ AGENT_USER }} -c "{{ CMD }}"
```
Comment (tehasdf) [inline diamond_agent/resources/diamond]: can we just `sudo -u {{ AGENT_USER }} {{ CMD }}`?

File: cloudify_handler/cloudify.py [hunk 2]
```diff
@@ -71,6 +72,7 @@ def _bind(self):
                                       internal=False)
 
     def process(self, metric):
+        global consec_err_count
```
Comment (tehasdf) [inline cloudify_handler/cloudify.py]: can this be just on self, an instance attribute? not global

File: cloudify_handler/cloudify.py [hunk 3]
```diff
@@ -80,8 +82,16 @@ def process(self, metric):
...
+                self.log.info(
+                    "Failed publishing to rabbitMQ. Attempting reconnect")
+            # Reducing 90% of errors in logs when
+            # there is a lasting connection problem
+            elif consec_err_count == 9:
+                consec_err_count = 0
+
+            consec_err_count = consec_err_count + 1
             self._bind()
             sleep(3)
```
Comment (tehasdf) [inline cloudify_handler/cloudify.py]: btw maybe instead 3 we can make it longer - using exponential backoff, so it's first sleep(1), then 2, then 4, 8.. up to 30
like here https://github.com/cloudify-cosmo/cloudify-common/blob/master/cloudify/amqp_client.py#L107-L113

Comment (geokala) [reply cloudify_handler/cloudify.py]: Agreed- also, that might mean that the complexity of the consec_err_count logic isn't needed?

File: diamond_agent/resources/diamond (hunk 1 above)
Comment (geokala) [reply diamond_agent/resources/diamond]: That would be better.

File: diamond_agent/tasks.py [hunk 4]
```diff
@@ -356,7 +356,7 @@ def create_config(path_prefix,
             'class': 'handlers.TimedRotatingFileHandler',
             'level': 'DEBUG',
             'formatter': 'default',
-            'args': "('{0}', 'midnight', 1, 7)".format(
+            'args': "('{0}', 'midnight', 1, 2)".format(
```
Comment (geokala) [inline diamond_agent/tasks.py]: How necessary is this with a sensibly back-off algorithm as discussed above?

**Pull request merged**